### ISSUE TITLE
[Feature-Request] A simple child window per task

### ISSUE THREAD

**Issue by theweavrs** [agalwood/Motrix#311]
**Is your feature request related to a problem? Please describe.**
Currently, you have to maximize whole Motrix app to check the progress of a task. This is overkill.

**Describe the solution you'd like**
It would be very helpful, if we had a simple child window per task where we could control and check progress of an individual task. This is similar to how other download managers do it e.g. IDM, XDM etc.

**So there will be 1 main window (a dashboard) and multiple child windows for each task.**

**Describe alternatives you've considered**
None.

**Additional context**
None.

Would this be too much memory/performance wise? Is it feasible do you think? Does the current architecture of the project support this or will there be need of too many changes?

**Comment by agalwood** [comment]
I think the polling of the task progress is a problem.

**Comment by theweavrs** [comment]
@theweavrs can we poll for individual tasks or are all tasks fetched together always?

**Comment by theweavrs** [comment]
@agalwood we can use the MainWindow as a relay to pass the updated progress to the child windows. This will work because each task will always have its window (unless you want to make it optional).

**Comment by agalwood** [comment]
I think the electron integrated chromium is a memory monster. Each window is equivalent to starting a process, and the memory pressure will be a bit high.

**Comment by theweavrs** [comment]
That is true. However, if we do it smartly we can make this possible without too much of a difference. Consider a task window; it will only have a progress bar, some text boxes and a few buttons. Nothing too complex will happen inside the task window. I know that no matter how simple it is, a process is a process but VSCode also does it in the form of tabs (I think) although it does have memory issues from time to time, I think we can agree that VSCode is far more functionally complex than our simple Motrix.

I think we should give it a try as a test to see how much memory-heavy it becomes. If it's too much we can exclude it or set it as an option.

What do you think?
